### PR TITLE
feat(combobox, dropdown, input-date-picker): Add flipPlacements property

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #2814

## Summary

feat(combobox, dropdown, input-date-picker): Add flipPlacements property. #2814

Expose property for fallback placements

### CODE REVIEW

**Review by eriklharper — APPROVED** [review]
LGTM! 👍

**Comment by driskull** [comment]
@jcfranco can you approveth?

File: src/utils/popper.ts [hunk 1]
```diff
@@ -22,10 +23,96 @@ type VariationRtl =
   | "left-leading"
   | "left-trailing";
 
+export type PopperComputedPlacement = ComputedPlacement;
```
Comment (jcfranco) [inline src/utils/popper.ts:26]: Can you drop `Popper` from this interface to hide the implementation details? 

We may need to introduce another type to replace `PopperPlacement` too (this one is a separate item).

**Review by jcfranco — APPROVED** [review]
Aside from the rename comment, this looks good to me! 🎉

File: src/utils/popper.ts (hunk 1 above)
Comment (driskull) [reply src/utils/popper.ts:26]: Yeah we can rename all of these. Maybe when we move to floatingUI?

**Pull request merged**